src/cart/crt_context.c

```diff
@@ -370,19 +350,90 @@ crt_context_provider_create(crt_context_t *crt_ctx, crt_provider_t provider, boo
 bool
 crt_context_is_primary(crt_context_t crt_ctx)
 {
-	struct crt_context *ctx;
-
-	ctx = crt_ctx;
+	struct crt_context *ctx = crt_ctx;
 
 	return ctx->cc_primary;
 }
 
 int
 crt_context_create(crt_context_t *crt_ctx)
 {
-	return crt_context_provider_create(crt_ctx, crt_gdata.cg_primary_prov, true);
+	return crt_context_provider_create(crt_ctx, crt_gdata.cg_primary_prov, true, 0);
+}
+
+uint32_t
+crt_num_ifaces_get(void)
+{
+	return crt_provider_num_ifaces_get(true, crt_gdata.cg_primary_prov);
 }
 
+int
+crt_context_create_on_iface_idx(uint32_t iface_index, crt_context_t *crt_ctx)
+{
+	uint32_t num_ifaces;
+
+	if (crt_is_service()) {
+		D_ERROR("API not available on servers\n");
+		return -DER_NOSYS;
+	}
+
+	num_ifaces = crt_num_ifaces_get();
+	if (num_ifaces == 0) {
+		D_ERROR("No interfaces specified at startup\n");
+		return -DER_INVAL;
+	}
+
+	if (iface_index >= num_ifaces) {
+		D_ERROR("interface index %d outside of range [0-%d]\n",
+			iface_index, num_ifaces - 1);
+		return -DER_INVAL;
+	}
+
+	return crt_context_provider_create(crt_ctx, crt_gdata.cg_primary_prov, true, iface_index);
+}
+
+int
+crt_iface_name2idx(char *iface_name, int *idx)
+{
+	uint32_t	num_ifaces;
+	int		i;
+	char		*name;
+	int		rc = -DER_INVAL;
+
+	num_ifaces = crt_provider_num_ifaces_get(true, crt_gdata.cg_primary_prov);
+
+	for (i = 0; i < num_ifaces; i++) {
+		name = crt_provider_iface_str_get(true, crt_gdata.cg_primary_prov, i);
+
+		if (strcmp(name, iface_name) == 0) {
+			*idx = i;
+			rc = DER_SUCCESS;
+			break;
```

Comment:
you could directly return without the need for `rc`.

Reply:
sure, will simplify if i do another push

Reply:
was that pushed ? I did not see it

Reply:
no, i didnt push that one:) i will if there are more changes to make, otherwise would like to avoid another ci run

Reply:
done